modules/core/src/main/java/org/apache/ignite/internal/processors/datastreamer/DataStreamerImpl.java

```diff
@@ -204,6 +204,10 @@
     /** Cancellation reason. */
     private volatile Throwable cancellationReason = null;
 
+    /** Cancellation error. */
+    private IgniteCheckedException cancellationError = new IgniteCheckedException
```

Comment:
No need to create exception if it isn't needed. Create it when it should be used.